### PR TITLE
feat: make query_stdio_capabilities work for windows

### PR DESCRIPTION
**Pull request by TerakomariGandesblood**
- Since rustix is already an integral part of running on non-Windows, remove it from features and make it a non-Windows dependency.
- Extracted several platform-dependent functions to reuse most of the query_stdio_capabilities code.
- Fixed a bug in Parser that incorrectly recognized older versions of Windows Terminal as supporting sixel.
- Temporarily skipped one of the doc tests due to it not working correctly when running the test on Windows (it works fine when running the test on its own)

### CODE REVIEW

File: src/picker.rs [hunk 1]
```diff
@@ -406,11 +501,13 @@ impl Parser {
...
                 'c' => {
+                    self.data.push(next);
+
                     // This is just easier than actually parsing the string.
                     let is_sixel = self.data.contains(";4;")
                         || self.data.contains("?4;")
-                        || self.data.contains(";4")
-                        || self.data.contains("?4");
+                        || self.data.contains(";4c")
+                        || self.data.contains("?4c");
```
Comment (j4james) [inline src/picker.rs:510]: FYI, you shouldn't need to check the variants beginning with `?`. If the first parameter of the `DA` response is a `4`, that indicates that the terminal is a VT132 terminal, which did _not_ support sixel.

Technically the first parameter should be 60 or more if you follow the spec exactly, but there are a number of modern terminals that aren't spec compliant, so you're probably better off ignoring that particular requirement.

In short, I'd recommend checking from parameter 2 onwards, i.e. you only need to search for `";4;"` or `";4c"`.

Comment (TerakomariGandesblood) [reply src/picker.rs:510]: Thanks for the comment, I learned something new.
But here is just fixing a previous bug, I think simplifying this check is another issue that should be put into a new PR to fix.

**Review by benjajaja — COMMENTED** [review]
Thank you! I really wanted this to happen (even though I don't use windows myself), and a lot of people have been asking for windows support.

There are some small changes that I commented but I will fix them myself after merging, don't worry.

File: src/picker.rs [hunk 2]
```diff
@@ -266,17 +264,129 @@ fn iterm2_from_env() -> Option<ProtocolType> {
...
+    if let Some(in_mode) = ORIGINAL_IN_MODE.lock().unwrap().as_ref() {
+        unsafe { Console::SetConsoleMode(in_handle, *in_mode) }?;
+    }
+
+    Ok(())
+}
+
+#[cfg(windows)]
+fn font_size_fallback() -> Option<FontSize> {
+    Some((10, 20))
```
[the remark this answers is not in the document]
Comment (TerakomariGandesblood) [reply src/picker.rs:385]: Getting the font size through the API should be a better option. Unfortunately I didn't find a way to do this under Windows, but I think it's something that could be done. Of course in case the api fails, it would be nice to have a chance to specify the font size manually.

File: src/picker.rs [hunk 3]
```diff
@@ -266,17 +264,129 @@ fn iterm2_from_env() -> Option<ProtocolType> {
...
 
-#[cfg(feature = "rustix")]
-fn query_stdio_capabilities(is_tmux: bool) -> Result<(Option<ProtocolType>, Option<FontSize>)> {
-    use rustix::termios::{LocalModes, OptionalActions};
+#[cfg(not(windows))]
+static ORIGINAL_IN_MODE: std::sync::Mutex<Option<rustix::termios::Termios>> =
+    std::sync::Mutex::new(None);
+
+#[cfg(not(windows))]
+fn enable_raw_mode() -> Result<()> {
```
[the remark this answers is not in the document]
Comment (TerakomariGandesblood) [reply src/picker.rs:272]: There's another issue that needs to be addressed here, when enable_raw_mode is called, if the code behind it exits early with an error, it may cause disable_raw_mode not to be called. Here we need a scope guard or something like that.

[the remark this answers is not in the document]
Comment (benjajaja) [reply src/picker.rs:272]: Oh I didn't think of that! I will try to fix that somehow.
